lib/discourse.rb

```diff
@@ -412,7 +412,7 @@ def self.route_for(uri)
     unless uri.is_a?(URI)
       uri = begin
         URI(uri)
-      rescue URI::Error
+      rescue ArgumentError, URI::Error
```

Comment:
Just to clarify, when will URI result in an argument error?

Reply:
When the URI is `nil`.
```
> URI(nil)
ArgumentError: bad argument (expected URI object or URI string)
```

Reply:
In that case, I think we should just add `return nil if uri.blank?` to the top of this method so that we don't have to do extra work when we already know that `uri` is invalid.